src/envConfig.ts

```diff
@@ -52,12 +45,13 @@ export function envConfig(userOptions: Partial<EnvConfigOptions> = {}): Plugin {
       });
     },
 
-    closeBundle() {
+    generateBundle() {
       const templateContent = createEnvConfigContent(userOptions.variables || [], true);
-
-      const TEMPLATE_PATH = path.join(config.root, config.build.outDir, 'env-config.template.js');
-      fs.mkdirSync(path.dirname(TEMPLATE_PATH), { recursive: true });
-      fs.writeFileSync(TEMPLATE_PATH, templateContent, 'utf8');
+      this.emitFile({
+        type: 'asset',
+        fileName: 'env-config.template.js',
+        source: templateContent,
+      });
```

Comment:
How does this create the folder?

Reply:
It doesn't, it just registers the file for the bundle and rollup takes care of the rest.